### PR TITLE
✨  [issue-254] Migrate Home Page to use dynamic content from Strapi

### LINKED ISSUE #254: [feature] migrate home page to use dynamic content from strapi
Once https://github.com/Developer-DAO/cms/issues/6 has been merged, the home page will need to be update to replace the hard coded content with dynamic content from the CMS.

The model is being documented in the issue reference above but it should ultimately be a 1 for 1 change that requires:

- [ ] Adding GraphQL query to call in data in `getServerSideProps`
- [ ] Replacing hard content content with data from this query

### PR DESCRIPTION
**Pull request by andriishupta**
### What does it do?
Migrates HomePage to take dynamic data from Strapi CMS
- add SEO component and move it to HomePage
- fill data: heading / sub_heading / values / goals / mission / footer
- refactor components to match data from CMS
- create CMS types
- create homePageQuery

### Any helpful background information?
-

### Any new dependencies? Why were they added?
-

### Relevant screenshots/gifs
-

### Does it close any issues?
Closes #254, #247, #248

### CODE REVIEW

**Review by kempsterrrr — COMMENTED** [review]
@andriishupta  Great work again mate! Left one specifc comment in the reivew re changing the name of the Values component  > Purpose :)

Some quick questions:

1. Do we get the types from Strapi for free or are we writing these out ourselves? (If no, can we do this when we move to a monorepo?)
2. What was the reason for creating a global place for graphql quieries and types?

Wondering about the DX of extracting these into their own folders away from where they're being used. If things are being re-used this makes sense but I'm not sure it makes sense for the page queries for example as they're only going to be used on that page. Moving them like this means we've got to go to another folder to check the query rather than just see it in `page/index.tsx` where it is being used.

Wondering the same thing about the types as. Is it better for them to live in the file/folder where they're being used or not? If not, why?

Given this is will have a long-term impact on the codebase be good to have a discussion and get @mhmulford0 thoughts as well :)

**Comment by andriishupta** [comment]
0) Thanks!
1) Currently, strapi is in JS, but we can move it to TS and auto-generate types AND use it in monorepo - agree with this
2) Personal preference AND in future I would like to import all types from CMS GraphQL - usually it is 1 schema file
Could update if needed 
3) Values contained: values, goals, mission - why Values and not Goals? Maybe Mission?) So I chose general word to combine 3 of those
4) folder creation - personal preference and experience from previous projects
Same for folder Components - always hated React's components, cause why I need to move specific component if I only use it in HomePage - like Purpose/Values
So this could be discussed :)

5) general code and approaches - subject to discuss, cause different guys used different approaches and we could agree on something)

File: src/Components/Purpose/index.tsx [hunk 1]
```diff
@@ -171,4 +147,4 @@ const Values = () => {
     </Flex>
   );
 };
-export default Values;
+export default Purpose;
```
[the remark this answers is not in the document]
Comment (andriishupta) [reply src/Components/Purpose/index.tsx:150]: answered in general comment - could revert or pick other name, such as Values contains values, goals, mission - what is the best single word for our Purpose?(i googled variants)

[the remark this answers is not in the document]
Comment (kempsterrrr) [reply src/Components/Purpose/index.tsx:150]: I clearly wasn't paying enough attention when looking at the component and just saw it as Values, Purpose sounds fine if it all of those things :)

**Comment by kempsterrrr** [comment]
> Currently, strapi is in JS, but we can move it to TS and auto-generate types AND use it in monorepo - agree with this

yes please! Feels like maybe we should start thiking about the migartion to turbo repo sooner rather than later. I noticed @mhmulford0 just shared a repo with a new version built here - https://github.com/mhmulford0/dd-website-monorepo - is anyone motivated to attempt this? Think I'll be short on time for the next couple of weeks for S1 launch

> Personal preference AND in future I would like to import all types from CMS GraphQL - usually it is 1 schema file
Could update if needed

Seems to make sense to me re anything coming from the CMS... are we expecting types on the FE that don't come from the CMS?

> folder creation - personal preference and experience from previous projects
Same for folder Components - always hated React's components, cause why I need to move specific component if I only use it in HomePage - like Purpose/Values
So this could be discussed :)

responded above :)

> general code and approaches - subject to discuss, cause different guys used different approaches and we could agree on something)

Yeah this will be evolving. where possible scoping things to the components where they live has always to me felt like a good idea so people working on them have everything that need in that file and don't have to context switch more than necassary. If there is a way we could keep the components types scoped near the components and then input the types from the CMS from a central place to consume in the component types that would be awesome

**Comment by kempsterrrr** [comment]
Just noticed one other thing dude which I don't think was you but we might as well fix. The partner logo's are not centered, if you look at this picture the last logo on the right is not alinged with the edge of the container.

Screenshot 2022-08-06 at 13 25 00

**Comment by mhmulford0** [comment]
The TS support for strapi just landed so that is an option. having the generics, for now, is suitable for a bit of dx and can be phased out as we bring the repos together. 

The work done for typing is great @andriishupta! Everything else looks good on that front. 

as far as folder structure I have always been a fan of components folder and having the things near their relevant parts. As kemp said, having the query, for example, broken out makes for more hunting a dev has to do around the code base. another would be the footer. typically I would define that as an interface inside the footer component

a component with many lines doesn't mean it's _too_ long necessarily, even more so if it's a one-off like with the homepage. 

For more generic types, it makes sense for them to live on their own like with `types/cms/general` `types/cms/strapi`

so a bit of both. i like my components to have typings in them, and higher-level stuff on its own.

**Comment by andriishupta** [comment]
@kempsterrrr, yes, we can agree that we should use page types somewhere close to it:
- everything inside the page itself, cause in Next.js we won't create new files inside /pages
- src/%PAGE%/components | types | graphql, etc.

centering images - would take a look and align them

@mhmulford0 
typings - thanks)
with monorepo + TS in CMS - we should use generated types

Yes, we can move them now, but after monorepo, we possibly would delete all manual typings cause we would get them from @cms/generated/types(or something like this) - I didn't expect to monorepo be soon, and manual creation of typing didn't take a lot of time

---

to summarize, I would:
- center images in this PR
-❓won't move typings cause if monorepo would be next week - we would delete them, so they don't matter
- will move the Graphql query back to the home page not to create a new folder structure


off-topic: for Graphql I vision generating types in this way, that is why I moved it in one place: https://www.graphql-code-generator.com/plugins/typescript/typescript-react-apollo

now we don't need it

**Comment by mhmulford0** [comment]
sounds like a plan. honestly, I didn't expect it to happen so soon either! I just gave it a shot and it went well. 

I think we are good on this. feel free to move this to active PR when you are ready.

**Review by mhmulford0 — APPROVED** [review]

**Comment by kempsterrrr** [comment]
this all sounds reasonable to me. haven't review as time sucks today but support merging whenever you're ready 🤝
